examples/with-eip-1193-provider/src/lib/turnkey.ts

```diff
@@ -0,0 +1,176 @@
+"use server";
+
+import {
+  ApiKeyStamper,
+  type TApiKeyStamperConfig,
+} from "@turnkey/api-key-stamper";
+import { TurnkeyClient, createActivityPoller } from "@turnkey/http";
+import { Attestation, Email, PassKeyRegistrationResult } from "./types";
+
+import { generateRandomBuffer, base64UrlEncode } from "./utils";
+import {
+  ETHEREUM_WALLET_DEFAULT_PATH,
+  PUBKEY_CRED_TYPE,
+  ALG_ES256,
+} from "./constants";
+import { UUID } from "crypto";
+import { Address } from "viem";
+
+const {
+  TURNKEY_API_PUBLIC_KEY,
+  TURNKEY_API_PRIVATE_KEY,
+  NEXT_PUBLIC_ORGANIZATION_ID,
+  NEXT_PUBLIC_TURNKEY_RPID,
+} = process.env;
+
+export const registerPassKey = async (
+  email: Email
+): Promise<PassKeyRegistrationResult> => {
+  if (!NEXT_PUBLIC_TURNKEY_RPID) {
+    throw "Error must define NEXT_PUBLIC_TURNKEY_RPID in your .env file";
+  }
+
+  // @todo - Add error handling
+  const { getWebAuthnAttestation } = await import("@turnkey/http");
+  const challenge = generateRandomBuffer();
+  const authenticatorUserId = generateRandomBuffer();
+  const user = email.split("@")[0];
+  // An example of possible options can be found here:
+  // https://www.w3.org/TR/webauthn-2/#sctn-sample-registration
+  const attestation = await getWebAuthnAttestation({
+    publicKey: {
+      rp: {
+        id: NEXT_PUBLIC_TURNKEY_RPID,
+        name: "Tunkey Demo Wallet",
+      },
+      challenge,
+      pubKeyCredParams: [
+        {
+          type: PUBKEY_CRED_TYPE,
+          alg: ALG_ES256,
+        },
+      ],
+      user: {
+        id: authenticatorUserId,
+        name: user,
+        displayName: user,
+      },
+      authenticatorSelection: {
+        requireResidentKey: true,
+        residentKey: "required",
+        userVerification: "preferred",
+      },
+    },
+  });
+
+  return { challenge: base64UrlEncode(challenge), attestation };
+};
+
+export const createAPIKeyStamper = (options?: TApiKeyStamperConfig) => {
+  const apiPublicKey = options?.apiPublicKey || TURNKEY_API_PUBLIC_KEY;
+  const apiPrivateKey = options?.apiPrivateKey || TURNKEY_API_PRIVATE_KEY;
+
+  if (!(apiPublicKey && apiPrivateKey)) {
+    throw "Error must provide public and private api key or define API_PUBLIC_KEY API_PRIVATE_KEY in your .env file";
+  }
+
+  return new ApiKeyStamper({
+    apiPublicKey,
+    apiPrivateKey,
+  });
+};
+
+export const createUserSubOrg = async (
+  turnkeyClient: TurnkeyClient,
+  {
+    email,
+    // if challenge and attestation are provided we are creating a custodial wallet using the users provided authenticator
```

Comment:
should this be `non-custodial` here? my understanding of this is: 
- if challenge and attestation are provided, the end-user is creating a suborg with their passkey == non-custodial
- if those are not provided, then we're adding an API key of the parent org == custodial